relayer/src/supervisor.rs

```diff
@@ -92,6 +92,37 @@ impl Supervisor {
                     }
                 }
 
+                IbcEvent::OpenInitConnection(..) | IbcEvent::OpenTryConnection(..) => {
+                    if !self.handshake_enabled() {
+                        continue;
+                    }
+
+                    let object = event
+                        .connection_attributes()
+                        .map(|attr| Object::connection_from_conn_open_events(attr, src_chain));
+
+                    if let Some(Ok(object)) = object {
+                        collected.per_object.entry(object).or_default().push(event);
+                    }
+                }
+
+                IbcEvent::OpenAckConnection(..) => {
+                    if !self.handshake_enabled() {
+                        continue;
+                    }
+
+                    if let Ok(connection_object) = Object::connection_from_conn_open_events(
+                        event.clone().connection_attributes().unwrap(),
```

Comment:
Is this `unwrap` guaranteed to succeed?

Also, the `clone` might not be required:

```suggestion
                        event.connection_attributes().unwrap(),
```

Reply:
> Is this unwrap guaranteed to succeed?

Yes, but we should probably follow the same pattern as for channel and ignore if not present (shouldn't happen but it's nicer)